src/components/OptionsList/BaseOptionsList.js

```diff
@@ -77,6 +80,63 @@ class BaseOptionsList extends Component {
         Log.hmmm('[OptionsList] scrollToIndex failed', info);
     }
 
+    /**
+     * This function is used to deterministically compute the layout of any given item in our list.
+     * We need to implement it so that we can programmatically scroll to any item in the list.
+     * If we don't, then we can only scroll to items in the virtual render window of the SectionList.
+     *
+     * @param {Array} data - This is the same as the data we pass into the component, i.e: all the non-empty sections
+     * @param {Number} index - This index is provided by React Native, and refers to a flat array with data from all the sections. This flat array has some quirks:
+     *
+     *     1. It ALWAYS includes a list header and a list footer, even if we don't provide/render those.
+     *     2. Each section includes a header, even if we don't provide/render one.
+     *
+     *     For example, given a list with two sections, two items in each section, no header, no footer, and no section headers, the flat array might look something like this:
+     *
+     *     [{header}, {sectionHeader}, {item}, {item}, {sectionHeader}, {item}, {item}, {footer}]
+     *
+     * @returns {Object}
+     * @note This implementation assumes that the list does not contain any headers or footers, because this component does not use those features at the time of writing.
+     *       If you want to add a header or footer to the SectionList, you'll need to adjust this function accordingly.
+     */
+    getItemLayout(data, index) {
+        const optionHeight = this.props.optionMode === CONST.OPTION_MODE.COMPACT ? variables.optionRowHeightCompact : variables.optionRowHeight;
+
+        let offset = 0;
+        let length = 0;
+        for (let i = 0; i < data.length; i++) {
+            const section = data[i];
+            const sectionHeaderHeight = section.title && !this.props.hideSectionHeaders ? variables.optionsListSectionHeaderHeight : 0;
+
+            // The indexOffset of our section is the sum of the lengths of previous sections' data arrays.
+            // To correct it, in the context of the flat array described above, we need to account for the section headers of all previous sections, and the list header.
+            const correctedIndexOffset = section.indexOffset + i + 1;
```

Comment:
Ok I think what I'm missing is that you are using `i + 1` here to mean `num section headers` so if we had index 1 we can assume there are 2 section headers etc.

This might sound tedious, but I think the code would be more obvious if we did something like this:

```js
const sectionHeaderCount = i + 1;
const correctedIndexOffset = section.indexOffset + sectionHeaderCount;
```